lib/battle/CBattleInfoCallback.cpp

```diff
@@ -855,6 +855,28 @@ AccessibilityInfo CBattleInfoCallback::getAccesibility() const
 		ret[BattleHex(0, y)] = EAccessibility::SIDE_COLUMN;
 	}
 
+	//special battlefields with logically unavailable tiles
+	std::vector<BattleHex> impassableHexes;
+	switch(battleGetBattlefieldType().num)
+	{

```

Comment:
It would, but switch will better reflect the purpose in case of adding more battlefields. Will be visible at first glance that whole control statement is about handling different battlefield types.